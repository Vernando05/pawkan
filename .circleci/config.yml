commands:
  build:
    description: |
      Create a production build
    steps:
    - run:
        command: mkdir -p artifacts
        name: Ensure artifacts directory is present
    - run: echo "$SERVERCRT" | base64 -d >> server.crt
    - run: echo "$SERVERKEY" | base64 -d >> server.key
    - run: NODE_ENV=production npm run build
    - run: cp -r dist artifacts
  test:
    description: |
      Test using Jest and Cypress
    steps:
    - run: echo "$SERVERCRT" | base64 -d >> server.crt
    - run: echo "$SERVERKEY" | base64 -d >> server.key  
    - run: npm run test:unit -- --collectCoverage
    - run: npm run test-ci
    - codecov/upload:
        file: coverage/clover.xml
  deploy:
    description: |
      Deploy to Firebase hosting
    steps:    
    - run: ./node_modules/.bin/firebase deploy --token "$FIREBASE_TOKEN" --only hosting
  lint:
    description: |
      Lint
    steps:
    - run: npm run lint
  install:
    description: |
      Install dependencies
    steps:
    - run: npm ci
orbs:
  codecov: codecov/codecov@1.1.3
executors:
  default:
    description: |
      Node 12 runtime environment
    docker:
    - image: cypress/browsers:<<parameters.cypress>>
    parameters:
      cypress:
        default: node12.13.0-chrome78-ff70
        type: string
jobs:
  build:
    description: |
      Build the application in production mode. The built assets are stored in artifacts.
    executor: default
    steps:
    - attach_workspace:
        at: ~/
    - build
    - persist_to_workspace:
        paths:
        - project   
        root: ~/
    - store_artifacts:
        path: ./artifacts
  test:
    description: |
      Run the tests and generate coverage reports.
    executor: default
    steps:
    - attach_workspace:
        at: ~/
    - test
  deploy:
    description: |
      Deploy to production
    executor: default
    steps:
    - attach_workspace:
        at: ~/
    - deploy
  lint:
    description: |
      Run lint on the source code
    executor: default
    steps:
    - attach_workspace:
        at: ~/
    - lint
  install:
    description: |
      Install dependencies. Checkout is called before install. (Optional) To prevent checkout from being called, pass your steps to before_install.
    executor: default
    parameters:
      before_install:
        default:
        - checkout
        description: |
          Steps to be executed before restoring cache and running install. Default is to run 'checkout' command.
        type: steps
    steps:
    - steps: <<parameters.before_install>>
    - restore_cache:
        key: v1-node-modules-{{ checksum "package-lock.json" }}
    - install
    - persist_to_workspace:
        paths:
        - project
        - .cache/Cypress
        root: ~/
    - save_cache:
        key: v1-node-modules-{{ checksum "package-lock.json" }}
        paths:
        - ~/.npm
        - ~/.cache
version: 2.1
workflows:
  version: 2.1
  front-test-build-deploy:
    jobs:
    - install
    - test:
        requires:
        - install
    - build:
        filters:
          branches:
            only:
            - dev
        requires:
        - test
    - deploy:
        filters:
          branches:
            only:
            - dev
        requires:
        - build
