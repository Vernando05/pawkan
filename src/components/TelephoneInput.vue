<template>
  <div :ref="telId" class="mstyle mb-2">
    <label v-if="label" :for="telId" class="v-label theme--light d-block mt-5 mb-2">{{ label }}</label>
    <vue-tel-input
      :inputId="telId"
      v-model="mobilePhone"
      @input="onInput"
      @validate="onValidate"
      @blur="onBlur"
      :preferredCountries="preferredCountries"
      :dynamicPlaceholder="dynamicPlaceholder"

      class="mstyle-field"
    ></vue-tel-input>
    <p :class="[errorMessage? '' : 'invisible', 'v-text-field__details caption error--text mb-0']">{{ errorMessage }}</p>
  </div>
</template>

<script lang="ts">
  import Vue from 'vue'
  import { VueTelInput } from 'vue-tel-input'
  import startCase from 'lodash/startCase'
  
  let telId = 0

  export default Vue.extend ({
    name: 'TelephoneInput',    
    components: {
      VueTelInput
    },
    props: {
      label: { type: String, required: false },
      preferredCountries: { type: Array, required: false },
      dynamicPlaceholder: { type: Boolean, required: false },
      value: { type: String, required: false, default: ''}
    },
    data () {
      return {
        telId: '' as string,
        mobilePhone: '' as string,
        isMobileNumberValid: false as boolean,
        isMobilePhoneDirty: false as boolean
      }
    },
    computed: {
      errorMessage (): string {
        if (this.mobilePhone.length && !this.isMobileNumberValid) {
          return this.$t('phoneNumberIsInvalid')
        } else
        if (!this.mobilePhone.length && this.isMobilePhoneDirty) {
          return startCase(this.$t('phoneNumber')) + ' ' + this.$t('cannotBeEmpty').toLowerCase()
        } else {
          return ''
        }
      }
    },
    created () {
      this.telId = 'telInput' + telId.toString()
      telId += 1
      this.mobilePhone = this.value
    },
    methods: {
      onBlur (): void {
        this.isMobilePhoneDirty = true
        if (!this.mobilePhone.length || !this.isMobileNumberValid) {
          (this.$refs[this.telId] as InstanceType<typeof HTMLDivElement>).classList.add('mstyle-field-error')
        } else {
          (this.$refs[this.telId] as InstanceType<typeof HTMLDivElement>).classList.remove('mstyle-field-error')
        }
      },
      onInput (number: string, { number: { e164 } }: { number: { e164: string } }): void {
        this.isMobilePhoneDirty = true
        this.mobilePhone = e164 || ''
        this.$emit('input', e164)
      },
      onValidate ({ isValid }: { isValid: boolean }): void {
        this.isMobileNumberValid = isValid
        this.$emit('validate', isValid)
      }
    }
  })
</script>
