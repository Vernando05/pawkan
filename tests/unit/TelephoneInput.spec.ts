import { renderComponent, flush } from './config'
import TelephoneInput from '@/components/TelephoneInput.vue'
import { fireEvent } from '@testing-library/vue'
import snapshotDiff  from 'snapshot-diff'
require('es6-promise').polyfill()
require('isomorphic-fetch')

jest.useFakeTimers()

test('telephone input', async () => {
  const { container, getByLabelText, queryByText, getAllByText, getByText } = renderComponent(TelephoneInput, {
    props: { 
      label: 'mobile',
      preferredCountries: ['id', 'sg'],
      dynamicPlaceholder: true
    }
  })

  const pristine = container.cloneNode(true)
  
  expect(getByLabelText(/mobile/i)).toBeInTheDocument()
  expect(queryByText(/'enter a phone number/i)).not.toBeInTheDocument()

  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  await fireEvent.touch(getByLabelText(/mobile/i))
  await flush()
  expect(queryByText(/empty/i)).toBeInTheDocument()

  expect(queryByText(/invalid/i)).not.toBeInTheDocument()
  await fireEvent.update(getByLabelText(/mobile/i), '+628123456')
  await flush()
  expect(queryByText(/empty/i)).not.toBeInTheDocument()
  expect(queryByText(/invalid/i)).toBeInTheDocument()

  await fireEvent.update(getByLabelText(/mobile/i), '+628123456789')
  await flush()
  expect(queryByText(/invalid/i)).not.toBeInTheDocument()

  await fireEvent.click(getByText(/â–¼/i))
  await flush()
  expect(getAllByText(/indonesia/i)[0]).toBeVisible()

  expect(snapshotDiff(pristine, container.cloneNode(true))).toMatchSnapshot()
})
