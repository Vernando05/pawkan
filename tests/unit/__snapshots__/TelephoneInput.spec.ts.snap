// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`telephone input 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,10 +1,10 @@
  <div
    data-app=\\"true\\"
  >
    <div
-     class=\\"mstyle mb-2\\"
+     class=\\"mstyle mb-2 mstyle-field-error\\"
    >
      <label
        class=\\"v-label theme--light d-block mt-5 mb-2\\"
        for=\\"telInput0\\"
      >
@@ -13,32 +13,32 @@
       
      <div
        class=\\"mstyle-field vue-tel-input\\"
      >
        <div
-         class=\\"vti__dropdown\\"
+         class=\\"vti__dropdown open\\"
          tabindex=\\"0\\"
        >
          <span
            class=\\"vti__selection\\"
          >
            <div
-             class=\\"vti__flag\\"
+             class=\\"vti__flag id\\"
            />
             
            <!---->
             
            <span
              class=\\"vti__dropdown-arrow\\"
            >
-             ▼
+             ▲
            </span>
          </span>
           
          <ul
            class=\\"vti__dropdown-list below\\"
-           style=\\"display: none;\\"
+           style=\\"\\"
          >
            <li
              class=\\"vti__dropdown-item preferred\\"
            >
              <div"
`;
